mathesar_ui/src/component-library/with-view-more-or-less-button/WithViewMoreOrLessButton.svelte

```diff
@@ -0,0 +1,50 @@
+<script lang="ts">
+  export let content: string;
+  export let maxLength: number;
+
+  let showComplete = false;
+
+  $: showExpandCollapseButton = content.length > maxLength;
+  $: completeText = content;
+  $: truncatedText = content.slice(0, maxLength);
+
+  function showCompleteView() {
+    showComplete = true;
+  }
+
+  function showTruncatedView() {
+    showComplete = false;
+  }
+</script>
+
+{#if showComplete}
+  <slot name="completeText" text={completeText} />
+  {#if showExpandCollapseButton}
+    <span
+      role="button"
+      class="expand-collapse-button"
+      on:click={showTruncatedView}
+    >
+      View Less
+    </span>
+  {/if}
```

Comment:
Move the showTruncatedView/showCompleteView logic out of the parent `if` block.

Reason: This span has a role of a button and so we need to allow the browser to be able to focus on the element. When the parent 'if' invalidates, it removes this span and renders the span from the 'else' block, which is a different element. This will lead to losing the focus on the element, which is an accessibility concern.